src/server/cea_608_encoder/byte_pair_generator.py

```diff
@@ -0,0 +1,107 @@
+import src.server.cea_608_encoder.caption_string_utility as utils
+import src.server.cea_608_encoder.scene_utility as scene_utils
+
+# At a future time this could live in it's own config file
+# Leaving it here temporarily
+supported_caption_formats = [
+    'CEA_608'
+]
+
+
+def consume(caption_data: dict) -> dict:
+    """Perform error handling around caption format and ensure
+    there are scenes to create byte pairs for.
+
+    :param caption_data: the full JSON blob from the front end
+    :return: TODO
+    """
+    if not caption_data['caption_format']:
+        raise ValueError('You must specify a caption format')
+
+    if caption_data['caption_format'] not in supported_caption_formats:
+        caption_format = caption_data['caption_format']
+        raise ValueError(f'The supplied caption format {caption_format} is not supported.')
+
+    if not caption_data['scenes_list']:
+        raise ValueError('Cannot encode byte pairs with an empty scene list.')
+
+    scene_data = caption_data['scenes_list']
+    caption_format = caption_data['caption_format']
+
+    return {
+        'type': caption_format,
+        'data': consume_scenes(scene_data)
+    }
+
+
+def consume_scenes(scene_list: list) -> list:
+    """Iterate over the list of scenes and create bytes for fields that
+    are set in the scene data. Call the consume function for caption
+    strings to return byte pairs for caption strings inside a scene.
+
+    :param scene_list:
+    :return: TODO
+    """
+    scene_data = []
+    for scene in scene_list:
+        if not scene['scene_id']:
+            raise ValueError('Every scene must have a scene ID.')
+
+        if not scene['start_time']:
+            raise ValueError('You must specify a starting time for a scene.')
+        else:
+            start_time = scene['start_time']
+            scene_utils.validate_time_formatting(start_time)
+
+        if scene['background_color']:
+            background_color = scene['background_color']
+            scene_utils.create_bytes_for_scene_background_color(background_color)
```

Comment:
Is this value supposed to be used somewhere? I know create_bytes_for_scene_background_color isn't defined yet, but I'm assuming it returns a value and you were intending to use that value somewhere.

Reply:
I believe things like background color are going to be doing bit manipulations on the "Wrapping" caption format data. 
The byte pairs that precede our text data will hold the background color information. So when parsing the style data you would start with a string of bits that are all 0 or all 1 for simplicity. 

0000 0000
Then when checking for background color if it was blue, the first 3 bits need to be 010 so you do a bit manipulation to make the byte pair look like this
0100 0000
or if it was green the first three bits need to be 111 so it looks like this instead
1110 0000

This is an example of what the function creare_bytes_for_scene_backround should do, this is not the exact way Byte pairs are changed with respects to backround color. @albicant  Should be able to confirm or deny this.

Reply:
It should definitely return a value, I'm not sure what format AWS Live expects the data to come in, once we know that we can start to piece together the return values from all these calls.